### PR TITLE
The creepy clown epidemic has arrived at Space Station 13.

### PR DESCRIPTION
**Pull request by harmonyn**
Implements a full clown event for Halloween to coincide with recent events. Suggestions on how to make it even more spooky are welcome.

:cl: bgobandit
add: The creepy clown epidemic has arrived at Space Station 13.
add: Honk.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Yesss space Halloween features

File: code/modules/events/holiday/halloween.dm [hunk 1]
```diff
@@ -53,6 +53,43 @@
...
+			return
+		H << "<span class='danger'>Honk...</span>"
+		H << 'sound/spookoween/scary_clown_appear.ogg'
+		new /obj/effect/hallucination/delusion(H.loc,H,force_kind="clown",duration=300,skip_nearby=0)
+
+/datum/round_event/creepy_clowns/tick()
+	if(IsMultiple(activeFor, 3))
+		for(var/mob/living/carbon/human/H in living_mob_list)
+			if (prob(66))
+				playsound(H.loc, pick('sound/spookoween/scary_horn.ogg','sound/spookoween/scary_horn2.ogg', 'sound/spookoween/scary_horn3.ogg'), 300, 1)
```
Comment (Xhuis) [inline code/modules/events/holiday/halloween.dm]: Wouldn't you want to play the sound directly to them? Also, volume caps at 100 for playback; having it any higher has no effect.

File: code/modules/events/holiday/halloween.dm [hunk 2]
```diff
@@ -53,6 +53,43 @@
...
+				new /obj/effect/hallucination/delusion(H.loc,H,force_kind="clown",duration=50,skip_nearby=0) //
+			else if (prob(33))
+				new /obj/effect/mob_spawn/human/corpse/clown(H.loc)
+			else if (prob(1))
+				new /mob/living/simple_animal/hostile/retaliate/clown(H.loc)
+
+/datum/round_event/creepy_clowns/announce()
+	priority_announce("Honk... Honk... HONK... HONK!", "HONK!", 'sound/spookoween/scary_horn.ogg')
+
+/datum/round_event/spooky/start()
```
Comment (Shadowlight213) [inline code/modules/events/holiday/halloween.dm]: ?

**Comment by Tacolizard** [comment]
please im so tired of creepy clown cancer

File: code/modules/events/holiday/halloween.dm (hunk 2 above)
Comment (Shadowlight213) [inline code/modules/events/holiday/halloween.dm]: Why is this code here? It looks like you accidentally left it after copying or something.
/datum/round_event/spooky/start() is already defined on line 9 with actual code.

Comment (bgobandit) [inline code/modules/events/holiday/halloween.dm]: I'm pretty sure that was there before, can double check though

Comment (optimumtact) [inline code/modules/events/holiday/halloween.dm]: how can it be there before when it's showing as something you added in the diff buddy

it doesn't do anything so you can safely delete it

Comment (optimumtact) [inline code/modules/events/holiday/halloween.dm]: Anyway fix that and I'll pull this in

Comment (optimumtact) [inline code/modules/events/holiday/halloween.dm]: actually I lied slightly, please use the 

    spawn_atom_to_turf(type, target, amount) 

proc for spawning the clowns

    spawn_atom_to_turf(/mob/living/simple_animal/hostile/retaliate/clown, H, 1)

Comment (optimumtact) [inline code/modules/events/holiday/halloween.dm]: well no it was an example, use it for all the spawns

Comment (optimumtact) [inline code/modules/events/holiday/halloween.dm]: oh I didn't see you passed arguments into one

instead do 

    var/turf/T = get_turf(H)
    if(T)
      spawning code..

File: code/modules/events/holiday/halloween.dm [hunk 3]
```diff
@@ -53,6 +53,43 @@
...
+/datum/round_event/creepy_clowns
+	endWhen = 40
+
+/datum/round_event/creepy_clowns/start()
+	for(var/mob/living/carbon/human/H in living_mob_list)
+		if(!H.client || !istype(H))
+			return
+		H << "<span class='danger'>Honk...</span>"
+		H << 'sound/spookoween/scary_clown_appear.ogg'
+		new /obj/effect/hallucination/delusion(H.loc,H,force_kind="clown",duration=300,skip_nearby=0)
```
Comment (AnturK) [inline code/modules/events/holiday/halloween.dm]: These are unfortunately blocking due to my short-sightedness, so you need to change this so it happens at once for everyone.

**Comment by bgobandit** [comment]
fixing

File: code/modules/events/holiday/halloween.dm [hunk 4]
```diff
@@ -53,6 +53,46 @@
...
+/datum/round_event/creepy_clowns
+	endWhen = 40
+
+/datum/round_event/creepy_clowns/start()
+	for(var/mob/living/carbon/human/H in living_mob_list)
+		if(!H.client || !istype(H))
+			return
+		H << "<span class='danger'>Honk...</span>"
+		H << 'sound/spookoween/scary_clown_appear.ogg'
+		var/turf/T = get_turf(H)
```
Comment (optimumtact) [inline code/modules/events/holiday/halloween.dm]: I don't think you need to check for a turf in this specific case

File: code/modules/events/holiday/halloween.dm [hunk 5]
```diff
@@ -53,6 +53,46 @@
...
+		if(T)
+			new /obj/effect/hallucination/delusion(H.loc,H,force_kind="clown",duration=300,skip_nearby=0)
+
+/datum/round_event/creepy_clowns/tick()
+	if(IsMultiple(activeFor, 3))
+		for(var/mob/living/carbon/human/H in living_mob_list)
```
Comment (JJRcop) [inline code/modules/events/holiday/halloween.dm]: It looks like you can just blindly cast to mob/living really, or even mob, you're not using any specific human vars

**Comment by nullbear** [comment]
make them start covered in blood.

**Pull request merged**